src/re.rs

```diff
@@ -9,13 +9,13 @@
 // except according to those terms.
 
 use std::borrow::Cow;
-use std::collections::HashMap;
-use std::collections::hash_map::Iter;
 use std::fmt;
 use std::ops::Index;
 #[cfg(feature = "pattern")]
 use std::str::pattern::{Pattern, Searcher, SearchStep};
 use std::str::FromStr;
+use std::collections::HashMap;
+use std::sync::Arc;
 
```

Comment:
Small nit, but I try to order imports by:

1. Category. (Group by `std`, then group by third party crates and then group by internal imports.)
2. Alphabetical.